### ISSUE TITLE
The demo app crashes when opening on the mobile phone

### ISSUE THREAD

**Issue by dspcad** [taichi-dev/taichi-aot-demo#124]
Just follow the instructions and generate the apk files.
I use the debug apk to install it on the phone.
However, the installed app flashes and then, crashes. 
Is there anything I missed? Thanks.

**Comment by PENGUINLIONG** [comment]
Could you provide more detail about your test platform? Like Android version (API level), vendor, chip version, etc. The things are continuously tested by our CI server so it's quite unlikely the Android build would break.

**Comment by dspcad** [comment]
Actually, I just run three scripts described in https://github.com/taichi-dev/taichi-aot-demo/tree/master/framework
- build-taichi-android.sh
- build-android.sh
- build-android-app.sh E5_sph

I couldn't use Android Studio to build because of the error "Cause: globalScope.projectBaseName must not be null"
My related environment variables are shown below:
TAICHI_C_API_INSTALL_DIR=/path/to/taichi-aot-demo/build-taichi-android-aarch64/install/c_api/
TAICHI_REPO_DIR=/path/to/taichi/
ANDROID_NDK_ROOT=/path/toAndroid/Sdk/ndk/25.2.9519653

Thanks.

**Comment by dspcad** [comment]
![oppo_phone](https://user-images.githubusercontent.com/22190436/225504987-e1df5ef0-0477-442e-9203-4c0b65cb5ff5.jpg)
Is the information above good for you?

**Comment by PENGUINLIONG** [comment]
That's funny because that's my test device (SDM855 is Snapdragon 855) and that's a quite robust platform. But honestly Android 9 is pretty old. Google added Vulkan support in Android 7 and things doesn't break so often only after a couple of years. So I think it's more likely a driver issue.

**Comment by dspcad** [comment]
If you would like to guide me on how to update the driver, that would be great :)
Would you like to provide the information about your test phone?
Many thanks for your quick reply.

**Comment by PENGUINLIONG** [comment]
Mine is an old Xperia 1. It has the same chip Snapdragon 855 but in Android 10 (or 11? I forgot). Android GPU driver is updated together with system updates (usually) so you might wanna check that. But sorry I can't give you more hint than that. Everything works well in my case lol.

**Comment by dspcad** [comment]
It is OK. Would you mind sharing the screenshot of running the app on your old Xperia 1 lol
I will give it a try on Android 10/11 and share my results if I succeed.

**Comment by PENGUINLIONG** [comment]
It should be the same as this https://github.com/taichi-dev/taichi-aot-demo/blob/master/ci/headless-truths/linux/vulkan/E5_sph_headless.bmp. It's not currently with me.

**Comment by dspcad** [comment]
The headless and glfw work for me so android one should look the same as them, right?

**Comment by PENGUINLIONG** [comment]
Yes. All demos in this repo use the same code to run Taichi kernels. The only difference is headless writes to images and GLFW & Android app presents to a GUI window.

**Comment by dspcad** [comment]
Hi I tried a new phone but still failed. Below is the info of my test phone.
I installed debug apk like E5_sph-debug.apk. Is it correct? I couldn't install the release version because of the error
db: failed to install E5_sph-release-unsigned.apk: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed collecting certificates for /data/app/vmdl1448463087.tmp/base.apk: Failed to collect certificates from /data/app/vmdl1448463087.tmp/base.apk: Attempt to get length of null array]

![oppo_phone2](https://user-images.githubusercontent.com/22190436/225697482-6fefe223-d56c-42dd-9fa9-f035c7c95c9a.jpg)
